sherpa/fit.py

```diff
@@ -951,6 +952,43 @@ def fit(self, statfunc, pars, parmins, parmaxes,
                                  statargs, statkwargs)
 
 
+# What is the best way to annotate the return value here?
+#
+def _add_fit_stats(outfile: Optional[Union[str, Path, WriteableTextFile]],
+                   clobber: bool,
+                   names: Sequence[str]):
+    """Handle creating the output file handle for the fit callback.
+
+    If there is a file handle or file specified then write the
+    header. The return value should be used as a context manager to
+    ensure that the handle is closed (when outfile is a string or a
+    path).
+
+    """
+
+    if outfile is None:
+        return nullcontext(None)
+
+    if isinstance(outfile, WriteableTextFile):
+        fh = outfile
+    else:
+        # os.path.isfile and open accepts strings and Path objects
+        if not clobber and os.path.isfile(outfile):
+            raise FitErr('noclobererr', str(outfile))
```

Comment:
Should 'noclobererr' be 'noclobbererr'?  Not a  lien on this PR but simple a question for future clean up. I note that it is also spelled this way in sherpa/utils/err.py.

Reply:
Yeah - I spent a bit of time trying to check I had that spelled correctly. I don't know the design decision that came up with the existing label!